ivy/functional/frontends/numpy/indexing_routines/inserting_data_into_arrays.py

```diff
@@ -28,3 +30,138 @@ def fill_diagonal(a, val, wrap=False):
     temp = ivy.flatten(a)
     temp[:end:step] = val
     a = ivy.reshape(temp, shape)
+
+
+class AxisConcatenator:
+    # allow ma.mr_ to override this
+    concatenate = staticmethod(np_frontend.concatenate)
+    makemat = staticmethod(np_frontend.matrix)
+
+    def __init__(self, axis=0, matrix=False, ndmin=1, trans1d=-1):
+        self.axis = axis
+        self.matrix = matrix
+        self.trans1d = trans1d
+        self.ndmin = ndmin
+
+    def __getitem__(self, key):
+        if not isinstance(key, tuple):
+            key = (key,)
+
+        # copy attributes, since they can be overridden in the first argument
+        trans1d = self.trans1d
+        ndmin = self.ndmin
+        matrix = self.matrix
+        axis = self.axis
+
+        objs = []
+        # dtypes or scalars for weak scalar handling in result_type
+        result_type_objs = []
+
+        for k, item in enumerate(key):
+            scalar = False
+            if isinstance(item, slice):
+                step = item.step
+                start = item.start
+                stop = item.stop
+                if start is None:
+                    start = 0
+                if step is None:
+                    step = 1
+                if ivy.is_complex_dtype(step):
+                    size = int(abs(step))
+                    newobj = np_frontend.linspace(start, stop, num=size).ivy_array
+                else:
+                    newobj = np_frontend.arange(start, stop, step).ivy_array
+                if ndmin > 1:
+                    newobj = np_frontend.array(
+                        newobj, copy=False, ndmin=ndmin
```

Comment:
Here can we try to call `self.ndim` instead of creating a copy of it🙂

Reply:
Not sure what you mean by that. ndmin is basically a parameter for minimum number of dimensions an array should be. If I am missing something can you please elaborate a bit @hirwa-nshuti

Reply:
What I meant is that this should we can still reference to `self.ndim` as long as the value remains unchanged instead of creating new copies of such same variable🙄

Reply:
@hirwa-nshuti To clarify if you are reffering to the following lines 
https://github.com/unifyai/ivy/blob/d355d7b2c946d87fc8163e72325894d927c3a2b4/ivy/functional/frontends/numpy/indexing_routines/inserting_data_into_arrays.py#L51-L54

This is necessary as they might get updated depending on the first input argument, like here 
https://github.com/unifyai/ivy/blob/d355d7b2c946d87fc8163e72325894d927c3a2b4/ivy/functional/frontends/numpy/indexing_routines/inserting_data_into_arrays.py#L91-L93